File: sdk/fsharp/src/HiTechCloudSDK/NetworkServices.fs
namespace HiTechCloud.SDK

/// Network Services: IP addresses and reverse DNS for services
type NetworkServicesResource(httpClient: HttpClient) =

    /// GET /api/service/{id}/ip - List Service IP Addresses
    member this.ListIps() =
        async {
            return! httpClient.getAsync("/api/service/{id}/ip")
        }

    /// GET /api/service/{id}/rdns - Get reverse DNS entries
    member this.GetRdns() =
        async {
            return! httpClient.getAsync("/api/service/{id}/rdns")
        }

    /// POST /api/service/{id}/rdns - Update reverse DNS entries
    member this.UpdateRdns() =
        async {
            return! httpClient.postAsync("/api/service/{id}/rdns")
        }

File: sdk/clojure/src/hitechcloud/sdk/network_services.clj
(ns hitechcloud.sdk.network_services
  (:require [hitechcloud.sdk.http-client :as http]))

;; Network Services: IP addresses and reverse DNS for services

(defn list_ips "GET /api/service/{id}/ip - List Service IP Addresses" [client & [params]] (http-get client "/api/service/{id}/ip" params))

(defn get_rdns "GET /api/service/{id}/rdns - Get reverse DNS entries" [client & [params]] (http-get client "/api/service/{id}/rdns" params))

(defn update_rdns "POST /api/service/{id}/rdns - Update reverse DNS entries" [client & [params]] (http-post client "/api/service/{id}/rdns" params))

